lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,499 @@ module.exports = {
                             messageId: "unusedVar",
                             data: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
+                                : getDefinedMessageData(unusedVar),
+                            suggest: [
+                                {
+                                    messageId: "removeVar",
+                                    data: {
+                                        varName: unusedVar.name
+                                    },
+                                    fix(fixer) {
+                                        const id = unusedVar.identifiers[0];
+                                        const parent = id.parent;
+                                        const parentType = parent.type;
+                                        const tokenBefore = sourceCode.getTokenBefore(id);
+                                        const tokenAfter = sourceCode.getTokenAfter(id);
+
+                                        /**
+                                         * get range from token before of a given node
+                                         * @param {ASTNode} node node of identifier
+                                         * @param {number} skips number of token to skip
+                                         * @returns {number} start range of token before the identifier
+                                         */
+                                        function getBeforeToken(node, skips) {
```

Comment:
This doesn't seem to describe what the function is actually doing. Maybe `getPreviousTokenStart` would be a better name?

Reply:
Done!